### ISSUE TITLE
Unable to report issue on webcompat.com using "Report as foolip"

### ISSUE THREAD

**Issue by foolip** [webcompat/webcompat.com#2557]
Starting from https://webcompat.com/issues/new?label=type-google (HTTPS) and being logged in to the site using my GitHub account, I see a "Report as foolip" button alongside the "Report Anonymously" button. Using that button doesn't work, the whole form is fading in/out (intentional, I presume) but nothing happens. In the Chrome devtools console:
```
Refused to send form data to 'http://webcompat.com/' because it violates the following Content Security Policy directive: "form-action 'self'".
```

In other words, a problem of mixing http and https.

**Comment by foolip** [comment]
I also tried from https://webcompat.com/ and using "Report Anonymously", same problem.

@miketaylr red alert?

**Comment by miketaylr** [comment]
dang. ok, i'll revert and investigate.

**Comment by miketaylr** [comment]
> dang. ok, i'll revert and investigate.

rather, deploy the last version.

**Comment by miketaylr** [comment]
> Refused to send form data to 'http://webcompat.com/'

Why are we trying to submit to the http endpoint...

**Comment by foolip** [comment]
Don't know if you already deployed, but I just tested on Firefox and the problem doesn't manifest there. So possibly this is also an interop issue :)

**Comment by foolip** [comment]
Interesting, my 3 attempts actually did successfully file issues:
- https://github.com/webcompat/web-bugs/issues/17906 ("Report as foolip" in Chrome)
- https://github.com/webcompat/web-bugs/issues/17907 ("Report Anonymously" in Chrome)
- https://github.com/webcompat/web-bugs/issues/17908 ("Report Anonymously" in Firefox)

I'll close them. Good news is that the type-google label worked!

**Comment by miketaylr** [comment]
@foolip yeah, just barely re-deployed. Unsure if it was before or after you tested though. 

https://github.com/webcompat/webcompat.com/commit/b1bfaf5236175246c4e8308667f82721673d9732 is when we added the form-action CSP bits.

(site note: staging.webcompat.com is a place where we can file test bugs, fyi --  can manually deploy patches or branches there independent from production).

**Comment by miketaylr** [comment]
cc @laghee (since this relates to the CSP stuff)

**Comment by foolip** [comment]
> staging.webcompat.com is a place where we can file test bugs, fyi

Oh, I'll do that next time :) Which repo do the bugs end up in?

**Comment by softvision-sergiulogigan** [comment]
Right here, @foolip 
https://github.com/webcompat/webcompat-tests/issues

**Comment by foolip** [comment]
Thanks @miketaylr, much appreciated!

**Comment by laghee** [comment]
@miketaylr Any chance it's related to flask's `url_for`? https://stackoverflow.com/questions/34802316/make-flasks-url-for-use-the-https-scheme-in-an-aws-load-balancer-without-mess  Maybe specifying `PREFERRED_URL_SCHEME` would change behavior?

**Comment by miketaylr** [comment]
OK, I've deployed the temporary bandaid patch (which just removes form-action) and deployed that, but we should try to figure out what the real bug is here... I'm still not 100% sure this isn't some strange Chrome bug. Testing with Charles Proxy, I didn't see any traffic trying to hit the http endpoint, or a redirect of any kind.

**Comment by karlcow** [comment]
Maybe some contexts here. 
https://github.com/w3c/webappsec-csp/issues/8
see [for example](https://github.com/w3c/webappsec-csp/issues/8#issuecomment-406463864)
> The current situation where Chrome/Safari block redirects and Firefox does not makes deploying form-action very tricky.

302 and 307 seem to have different behaviors btw.
